src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -269,13 +312,29 @@ def create_replica(cmd, client, store_name, name, location, resource_group_name=
                                replica_creation_parameters=replica_creation_params)
 
 
-def delete_replica(cmd, client, store_name, name, resource_group_name=None):
+def delete_replica(cmd, client, store_name, name, yes=False, resource_group_name=None):
     if resource_group_name is None:
         resource_group_name, _ = resolve_store_metadata(cmd, store_name)
 
-    return client.begin_delete(resource_group_name=resource_group_name,
-                               config_store_name=store_name,
-                               replica_name=name)
+    if not yes:
+        config_store_client = cf_configstore(cmd.cli_ctx)
+        config_store = show_configstore(
+            cmd, config_store_client, store_name, resource_group_name
+        )
+        replicas = list_replica(cmd, client, store_name, resource_group_name)
+
+        if config_store.sku.name.lower() == "premium" and len(list(replicas)) == 1:
+            user_confirmation(
+                "Deleting the last replica will disable geo-replication. When using the premium tier, it is recommended to have geo-replication enabled to take advantage of an increased SLA of 99.99%. The first replica created for a premium tier store is included. Do you want to continue with this operation?"
```

Comment:
It is recommended that a premium tier store have geo-replication enabled to take advantage of the 99.99% SLA. The first replica for a premium tier store comes at no additional cost.

Reply:
We usually avoid documenting SLA in clients. We have dedicated documentation for SLAs.

Reply:
I see, Although these were strings discussed by Jimmy and Joe. We have added them on portal as well. I can confirm with them on this.

Reply:
Perhaps, we can remove the numbers?

```
Deleting the last replica will disable geo-replication. It is recommended that a premium tier store have geo-replication enabled to take advantage of the improved SLA. The first replica for a premium tier store comes at no additional cost.
```

@avanigupta @ChristineWanjau Thoughts?

Reply:
This makes sense to me. Updated